### PR TITLE
properly stop receiver plugins

### PR DESCRIPTION
**Pull request by boriwo**

### CODE REVIEW

File: internal/pkg/plugin/manager.go [hunk 1]
```diff
@@ -261,55 +277,40 @@ func (m *manager) stopReceiving(ctx context.Context, r *receiver) error {
...
-	//r.active = false
-
 	if r.done == nil {
 		return &NotRegisteredError{}
 	}
-	//BW
-	//close(r.done)
-
+	if r.active {
+		close(r.done)
```
Comment (holidaymike) [inline internal/pkg/plugin/manager.go:286]: should we set r.active to false here?

File: pkg/sender/types.go [hunk 2]
```diff
@@ -46,4 +46,8 @@ type Sender interface {
 	// StopSending will stop any long running maintenance threads in the sender plugin.
 	// Often this function does nothing.
 	StopSending(ctx context.Context)
+	//
+	Config() interface{}
+	Name() string
+	Plugin() string
```
Comment (holidaymike) [inline pkg/sender/types.go:52]: Why doesn't Filter interface also have these three methods?

File: internal/pkg/plugin/manager.go (hunk 1 above)
Comment (boriwo) [reply internal/pkg/plugin/manager.go:286]: ok

File: pkg/sender/types.go (hunk 2 above)
Comment (boriwo) [reply pkg/sender/types.go:52]: i didn't have the patience to implement this for filters yet and i only really needed visibility into sender and receivers anyway, but you are right, will add the missing apis shortly

**Pull request merged**